rfcs/0078-configuration-file-generators.md

```diff
@@ -123,7 +126,15 @@ widespread use of the module system inside `pkgs/`. There is also no guarantee
 that all the configuration files describing interaction of multiple software
 packages will have a clear choice of reference package.
 
-There also have been many solutions proposed based on a significant rework of the module system.
+A complete or partial merge of the module collections of the major currently
+existing module system based code bases, with internal options for platform
+specific implementation details. Unfortunately, this would force more coupling
+of the development cycle for the platform-specific parts. It also does not
+create a clean inspection/override point, and mixes the code with different
+platform requirements for testing.
```

Comment:
Again, the common module isn't platform specific and can stand alone.
It's really just like a function, except you get to reuse the `option` declarations for its inputs.

The two approaches are isomorphic if it wasn't for the fact that going back to a simple function loses the option metadata and checking.

inputs <-> normal options
outputs <-> internal options
multiple invocations <-> submodules
function without free variables <-> module that only defines self-declared options
function with small closure <-> module that doesn't import much